drivers/SmartThings/zigbee-switch/src/multi-switch-no-master/init.lua

```diff
@@ -37,35 +51,97 @@ local function is_multi_switch_no_master(opts, driver, device)
         return true
       end
   end
-
   return false
 end
 
-local function component_to_endpoint(device, component_id)
-  local ep_num = component_id:match("switch(%d)")
-  return ep_num and tonumber(ep_num) + 1 or 1
+local function get_children_amount(device)
+  for _, fingerprint in ipairs(MULTI_SWITCH_NO_MASTER_FINGERPRINTS) do
+    if device:get_model() == fingerprint.model then
+      return fingerprint.children
+    end
+  end
+end
+
+local function device_added(driver, device, event)
+  if device.network_type == st_device.NETWORK_TYPE_ZIGBEE then
+    local children_amount = get_children_amount(device)
+    for i = 1, children_amount do
+      local name = string.format("%s %d", device.label, i)
+      local metadata = {
+        type = "EDGE_CHILD",
+        label = name,
+        profile = "multi-switch-no-master",
+        parent_device_id = device.id,
+        parent_assigned_child_key = string.format("%02X", i),
+        vendor_provided_label = name,
+      }
+      driver:try_create_device(metadata)
+    end
+  end
 end
 
-local function endpoint_to_component(device, ep)
-  local switch_comp = string.format("switch%d", ep - 1)
-  if device.profile.components[switch_comp] ~= nil then
-    return switch_comp
-  else
-    return "main"
+local function find_child(parent, ep_id)
+  return parent:get_child_by_parent_assigned_key(string.format("%02X", ep_id))
+end
+
+local function device_init(driver, device, event)
+  if device.network_type == st_device.NETWORK_TYPE_ZIGBEE then
+    device:set_find_child(find_child)
   end
 end
 
-local device_init = function(self, device)
-  device:set_component_to_endpoint_fn(component_to_endpoint)
-  device:set_endpoint_to_component_fn(endpoint_to_component)
+local function switch_on_command_handler(driver, device, command)
```

Comment:
do the defaults not work for zigbee either?

Reply:
yes

Reply:
that's not what I would expect, would you @varzac ? If the defaults are using this driver's `endpoint_to_component` mapping, everything should go to main and the dst endpoint should get set by passing the device to the command constructor

Reply:
I managed to eliminate switch_on_command_handler and switch_off_command_handler, now it is handled by defaults. The problem was in mapping parent's endpoint